Flank/src/main/java/com/walmart/otto/aggregator/ArtifactsProcessor.java

```diff
@@ -0,0 +1,233 @@
+
+  private static final String CUT_VIDEO_COMMAND_TEMPLATE = "ffmpeg -i %s -ss %s -to %s %s";
+
+  private Configurator configurator;
+
+  public ArtifactsProcessor(Configurator configurator) {
+    this.configurator = configurator;
+  }
+
+  public void generateArtifactsForTestCase(Path reportBaseDir, String matrixName, TestCase testCase)
+      throws IOException, InterruptedException {
+
+    final Pattern targetTestStartedLine = createTestStartedPattern(testCase);
+    final Pattern targetTestFinishedLine = createTestFinishedPattern(testCase);
+
+    Path sourceLogcatFile =
+        reportBaseDir.resolve(
+            testCase.getShardName() + File.separator + matrixName + File.separator + "logcat");
+
+    Path logcatsFolder = getOrCreateChildFolder(reportBaseDir, "logcat");
+    Path testLogcatFile = logcatsFolder.resolve(testCase.getId() + ".txt");
+
+    Path entireLogcatFile = logcatsFolder.resolve("shard_" + testCase.getShardName() + ".txt");
+    Files.copy(sourceLogcatFile, entireLogcatFile, StandardCopyOption.REPLACE_EXISTING);
+
+    Instant firstTestStartedTime = null;
+    Instant targetTestStartedTime = null;
+    Instant targetTestFinishedTime = null;
+
+    if (!Files.exists(sourceLogcatFile)) {
+      System.err.printf(
+          "Can't process logcat file for %s. Expected file %s does not exist.%n",
+          testCase.getId(), sourceLogcatFile.toString());
+      return;
+    }
+
+    try (BufferedReader reader = Files.newBufferedReader(sourceLogcatFile);
+        BufferedWriter writer = Files.newBufferedWriter(testLogcatFile, Charset.forName("UTF-8"))) {
+
+      String line;
+      boolean copy = false;
+
+      while ((line = reader.readLine()) != null) {
+
+        if (firstTestStartedTime == null && TEST_STARTED_LINE.matcher(line).matches()) {
+          firstTestStartedTime = extractInstant(line);
+        }
+
+        if (targetTestStartedLine.matcher(line).matches()) {
+          targetTestStartedTime = extractInstant(line);
+          copy = true;
+        }
+
+        if (copy) {
+          writer.write(line);
+          writer.newLine();
+        }
+
+        if (targetTestFinishedLine.matcher(line).matches()) {
+          targetTestFinishedTime = extractInstant(line);
+          break;
+        }
+      }
+    }
+
+    processVideo(
+        reportBaseDir,
+        matrixName,
+        testCase,
+        firstTestStartedTime,
+        targetTestStartedTime,
+        targetTestFinishedTime);
+  }
+
+  private void processVideo(
+      Path reportBaseDir,
+      String matrixName,
+      TestCase testCase,
+      Instant firstTestExecuted,
+      Instant startOfTargetTest,
+      Instant endOfTargetTest)
+      throws IOException, InterruptedException {
+
+    Path videoFile =
+        reportBaseDir.resolve(
+            testCase.getShardName() + File.separator + matrixName + File.separator + "video.mp4");
+
+    if (!Files.exists(videoFile)) {
+      System.err.printf(
```

Comment:
should this return an error code or throw an exception?

Reply:
Not sure, I would rather silently shallow it at this point. It's probably better than aborting the entire execution in case for some reason we don't get the video file.